src/prefect/engine/flow_runner.py

```diff
@@ -147,10 +147,12 @@ def initialize_run(  # type: ignore
         """
 
         # overwrite context parameters one-by-one
-        if parameters:
-            context_params = context.setdefault("parameters", {})
-            for param, value in parameters.items():
-                context_params[param] = value
+        context_params = context.setdefault("parameters", {})
```

Comment:
having the default `parameters` inside even when none where entered is a blessing